include/boost/compute/utility/program_cache.hpp

```diff
@@ -82,6 +82,25 @@ class program_cache : boost::noncopyable
         m_cache.clear();
     }
 
+    /// Clears persistent cache by deleting the boost_compute folder
+    void clear_persistent_cache()
+    {
+#ifdef BOOST_COMPUTE_USE_OFFLINE_CACHE
+        // Path delimiter symbol for the current OS.
+        static const std::string delim = boost::filesystem::path("/").make_preferred().string();
+
+        // Path to appdata folder.
+#ifdef WIN32
+        static const std::string appdata = detail::getenv("APPDATA") 
+            + delim + "boost_compute";
+#else
+        static const std::string appdata = detail::getenv("HOME")
+            + delim + ".boost_compute";
+#endif
+        boost::filesystem::remove_all(appdata);
```

Comment:
I can't remember, but will this through an exception if the offline cache directory hasn't been created yet? If so we should catch/ignore that.